lib/quarantine/rspec_adapter.rb

```diff
@@ -1,86 +1,99 @@
+          if RSpec.configuration.skip_quarantined_tests \
+            && Quarantine::RSpecAdapter.quarantine.test_quarantined?(example)
+            Quarantine::RSpecAdapter.quarantine.pass_flaky_test(example)
+          end
         end
       end
     end
-  end
 
-  # Purpose: binds quarantine to record failed and flaky tests
-  def bind_quarantine_record_tests(quarantine)
-    ::RSpec.configure do |config|
-      config.after(:each) do |example|
-        metadata = example.metadata
+    # Purpose: binds quarantine to record failed and flaky tests
+    def self.bind_quarantine_record_tests
+      ::RSpec.configure do |config|
+        config.after(:each) do |example|
+          metadata = example.metadata
 
-        # will record the failed test if is not quarantined and it is on it's final retry from the rspec-retry gem
-        quarantine.record_failed_test(example) if
-          RSpec.configuration.quarantine_record_failed_tests &&
-          !quarantine.test_quarantined?(example) &&
-          metadata[:retry_attempts] + 1 == metadata[:retry] && example.exception
+          # will record the failed test if is not quarantined and it is on it's final retry from the rspec-retry gem
+          Quarantine::RSpecAdapter.quarantine.record_failed_test(example) if
+            RSpec.configuration.quarantine_record_failed_tests &&
+            !Quarantine::RSpecAdapter.quarantine.test_quarantined?(example) &&
+            metadata[:retry_attempts] + 1 == metadata[:retry] && example.exception
 
-        # will record the flaky test if is not quarantined and it failed the first run but passed a subsequent run;
-        # optionally, the upstream RSpec configuration could define an after hook that marks an example as flaky in
-        # the example's metadata
-        quarantine.record_flaky_test(example) if
-          RSpec.configuration.quarantine_record_flaky_tests &&
-          !quarantine.test_quarantined?(example) &&
-          (metadata[:retry_attempts] > 0 && example.exception.nil?) || metadata[:flaky]
+          # will record the flaky test if is not quarantined and it failed the first run but passed a subsequent run;
+          # optionally, the upstream RSpec configuration could define an after hook that marks an example as flaky in
+          # the example's metadata
+          Quarantine::RSpecAdapter.quarantine.record_flaky_test(example) if
+            RSpec.configuration.quarantine_record_flaky_tests &&
+            !Quarantine::RSpecAdapter.quarantine.test_quarantined?(example) &&
+            (metadata[:retry_attempts] > 0 && example.exception.nil?) || metadata[:flaky]
+        end
       end
-    end
 
-    ::RSpec.configure do |config|
-      config.after(:suite) do
-        quarantine.upload_tests(:failed) if RSpec.configuration.quarantine_record_failed_tests
+      ::RSpec.configure do |config|
+        config.after(:suite) do
+          if RSpec.configuration.quarantine_record_failed_tests
```

Comment:
Super minor nit: consider making this an inline conditional or the next conditional not inline for consistency

Reply:
Oh, that was a Rubocop autocorrect. I'll fix it, thanks.

Reply:
Never mind, Rubocop won't let me do it. We'll tweak its config later.